bin/check-exercises.sh

```diff
@@ -0,0 +1,31 @@
+#!/usr/bin/env bash
+#
+# Runs tests only if the current Pull Request has modified a test or the root
+# CMakeLists.txt, or if we are running on the master branch.
+
+# Fail if any command fails
+set -e
+
+repo=$(cd "$(dirname "$0")/.." && pwd)
+
+# Configure all the tests.
+cmake -G Ninja "$repo"
+echo ""
+
+if [ "$TRAVIS_PULL_REQUEST" == "false" ]; then
+    echo "This is not a Pull Request, running all tests."
+    cmake --build "$repo"
+elif git diff --name-only master | grep "^CMakeLists.txt"; then
+    echo "This PR changes the root CMakeLists.txt."
+    echo "Running a single test and then all tests."
+    cmake --build "$repo" -- test_hello-world
+    cmake --build "$repo"
+else
+    tests="$(git diff --name-only master | grep "exercises/" | cut -d '/' -f2 | sed 's/.*/test_&/' | tr '\n' ' ')"
```

Comment:
Can we use `grep "^exercises/"` here, to be more precise?

Moreover, it's possible that multiple files being edited in the same exercise path, resulting in the same exercise name occurs multiple times. So I think we can add a `|uniq ` before piping into the `|tr` command.

Reply:
Good points all around. I've made both suggested changes :)

Neither changes the function of the script, but they do make it more clear. Thanks!

(CMake/Ninja ignore multiple targets in the build list)